File: exercise/kotlin/day00/src/main/kotlin/email/Encryption.kt
package email

import java.util.*
import javax.crypto.Cipher
import javax.crypto.spec.IvParameterSpec
import javax.crypto.spec.SecretKeySpec

data class Configuration(val key: String, val iv: String)

private const val AES = "AES"

class Encryption(private val configuration: Configuration) {
    fun encrypt(plainText: String): String =
        Base64.getEncoder().encodeToString(
            createCipher(Cipher.ENCRYPT_MODE).doFinal(
                plainText.toByteArray(Charsets.UTF_8)
            )
        )

    fun decrypt(encryptedText: String): String =
        String(
            createCipher(Cipher.DECRYPT_MODE).doFinal(
                decode(encryptedText)
            ), Charsets.UTF_8
        )

    private fun createCipher(mode: Int): Cipher =
        Cipher.getInstance("${AES}/CBC/PKCS5Padding")
            .also { c ->
                c.init(
                    mode,
                    SecretKeySpec(decode(configuration.key), AES),
                    IvParameterSpec(decode(configuration.iv))
                )
            }

    private fun decode(x: String): ByteArray? = Base64.getDecoder().decode(x)
}

File: exercise/php/day00/src/Email/Encryption.php
<?php

namespace Email;

use RuntimeException;

class Encryption
{
    private string $key;
    private string $iv;

    public function __construct(string $key, string $iv)
    {
        // Decode Base64-encoded key and IV
        $this->key = base64_decode($key);
        $this->iv = base64_decode($iv);
    }

    public function encrypt(string $plainText): string
    {
        // Encrypt using AES-256-CBC with PKCS7 padding
        $ciphertext = openssl_encrypt(
            $plainText,
            'AES-256-CBC',
            $this->key,
            OPENSSL_RAW_DATA, // Raw binary data
            $this->iv
        );

        if ($ciphertext === false) {
            throw new RuntimeException('Encryption failed.');
        }

        // Return Base64-encoded result to match C# behavior
        return base64_encode($ciphertext);
    }

    public function decrypt(string $encryptedText): string
    {
        // Decode the Base64-encoded ciphertext
        $ciphertext = base64_decode($encryptedText);

        // Decrypt using AES-256-CBC with PKCS7 padding
        $decrypted = openssl_decrypt(
            $ciphertext,
            'AES-256-CBC',
            $this->key,
            OPENSSL_RAW_DATA, // Raw binary data
            $this->iv
        );

        if ($decrypted === false) {
            throw new RuntimeException('Decryption failed.');
        }

        return $decrypted;
    }
}
